src/js/views/RaceCourseView.js

```diff
@@ -6,45 +6,62 @@ class RaceCourseView extends View {
   tag = "[RaceCourseView]";
 
   init() {
-    // console.log(`${this.tag}: init`)
-    this.on("start-racing", ({ detail: { runCount, cars } }) => {
-      
-      this.renderCars(cars);
+    this.on("start-racing", this.onStart);
+  }
 
-      cars = $("racing-car");
-      if (!Array.isArray(cars)) {
-        cars = [cars];
-      }
-      
-      while (runCount--) {
-        console.log(cars);
-        cars.forEach(car => console.log(car) || car.run(runCount));
-      }
-      
-      this.emit("finish-racing", {
-        result: cars.map(car => ({
-          name: car.name,
-          totalDistance: car.getTotalDistance()
-        }))
+  onStart = ({ detail: { runCount, carNames } }) => {
+    this.renderCars(carNames);
+
+    let carElements = $("racing-car");
+    if (!Array.isArray(carElements) && carElements instanceof HTMLElement) {
+      carElements = [carElements];
+    }
```

Comment:
```suggestion
    const carElements = [...$('racing-car')]
    //or
    const carElements = Array.from($('racing-car'))
```
$() 함수가 장단점이 있네요. 단일요소 또는 유사배열객체 두가지 타입으로 반환을 하다보니 반환하는쪽에서 체크를 해야하는것이 단점인것 같아요. 일단 자동차 이름 하나만 넣으면 에러가 나타는데 위에 input에서 이름 하나만 넣는걸 방지하는게 좋을꺼 같습니다. 그리고 이 코드라인에서 if문을 제거하고  유사배열객체를 배열로 바꿔주는 코드를 삽입하면 좀더 의미가 명확하지 않을까 생각되네욯

Reply:
맞아요 ㅜㅜ 배열 메서드를 쓸 때 단일 요소인 경우를 체크해야하는 단점이 있습니다. 
근데 위에 제시해주신 코드는 
- 이미 $()함수안에서 *단일요소가 아닌경우* 배열로 만들고 여기 15번째라인에서 다시 배열로 만들고 있고
```js
export const $ = (selector, baseElement = document) => {
  const selectors = [...baseElement.querySelectorAll(selector)];

  if (selectors.length > 1) return selectors;

  return selectors[0];
}
```
- *단일요소인 경우* 유사배열이 아니기 때문에 스프레드 연산자와 Array.from을 쓸 수 없어요.

Reply:
querySelectorAll 과 querySelector를 그냥 분리해서 사용해야겠다는 생각이드네요 ^^;;

Reply:
다음에는 분리해서 사용해보고 판단해도 좋을꺼 같네요 ^^ㅎㅎ